src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1317,6 +1320,14 @@ void RaftPart::processAskForVoteRequest(
         return;
     }
 
+    // If candidate's proposed term is same as current term and it is the leader we voted before,
+    // we could vote for it again because it maybe a retry
+    if (req.get_term() == term_ && leader_ == candidate) {
+        resp.set_error_code(cpp2::ErrorCode::SUCCEEDED);
+        resp.set_current_term(term_);
+        return;
+    }
+
```

Comment:
talked offline already, there will be a problem if : 

precondition
A: Term 14, logid 100
B: Term 14, logid 90
C: term 14, logid 95.

Then, Network broken for a while, A, B, C begin leader election

B get request of C before A. 

B: term = 15, leader_ = C, and won't vote for A forever.